### PR TITLE
feat(settings): allow to make a PR which changes both the settings and the data

### PR DESCRIPTION
**Pull request by Haroenv**
We need this change, because right now, if we merge a PR which changes the index settings, they are immediately applied to the production index, which could not have the correct data yet

Flow before this PR:

1. setSettings on production index
2. setSettings on bootstrap index
3. bootstrap
4. copy production settings to bootstrap index
5. move bootstrap index to production index
6. delete bootstrap index
7. replicate
8. watch

Flow after this PR

1. setSettings on bootstrap index
3. bootstrap (will set settings to prod if up to date)
5. move bootstrap index to production index with its settings
6. delete bootstrap index
7. replicate
8. watch

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -148,14 +149,13 @@ async function bootstrap(state) {
 
 async function moveToProduction() {
   log.info('🚚 starting move to production');
-  await client.copyIndex(c.indexName, c.bootstrapIndexName, [
```
Comment (Haroenv) [inline src/index.js]: we no longer copy the settings from the prod index to the bootstrap index, but rely on the bootstrap index being correct

File: src/index.js [hunk 2]
```diff
@@ -83,6 +83,7 @@ function infoDocs(offset, nbDocs, emoji) {
 
 async function bootstrap(state) {
   if (state.seq > 0 && state.bootstrapDone === true) {
+    await setSettings(mainIndex);
```
Comment (Haroenv) [inline src/index.js]: setSettings will always be called when bootstrap is done

File: src/index.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ const { index: mainIndex, client } = createAlgoliaIndex(c.indexName);
 const { index: bootstrapIndex } = createAlgoliaIndex(c.bootstrapIndexName);
 const stateManager = createStateManager(mainIndex);
 
-setSettings(mainIndex)
+Promise.resolve()
```
Comment (vvo) [inline src/index.js]: If only someone did a chore PR of moving to async/await... :D

File: src/index.js [hunk 4]
```diff
@@ -148,14 +149,13 @@ async function bootstrap(state) {
 
 async function moveToProduction() {
   log.info('🚚 starting move to production');
-  await client.copyIndex(c.indexName, c.bootstrapIndexName, [
+
+  const currentState = await stateManager.get();
+  await client.copyIndex(c.bootstrapIndexName, c.indexName, [
```
Comment (vvo) [inline src/index.js]: I believe we should write `await client.copyIndex(c.bootstrapIndexName, c.indexName)` which will copy the data + all settings (+synonyms, query rules).

The current line will not copy the data

Comment (Haroenv) [reply src/index.js]: oh you're right, good of you for catching that!

**Pull request merged**